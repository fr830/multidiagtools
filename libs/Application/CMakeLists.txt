# Project file for Multi Diag Tools

# Project
project(multidiagtools)

# List of source files
set(SOURCES_FILES 
    src/Mdt/Application.cpp
)

# Headers directories
include_directories("${CMAKE_CURRENT_SOURCE_DIR}/src")
include_directories("${CMAKE_CURRENT_SOURCE_DIR}/../QtSolutions/include")
# Because some headers are generated during building (config.h, Qt's UIC), we also build path
include_directories("${CMAKE_CURRENT_BINARY_DIR}")

# Output library name
set(LIBRARY_NAME "Application")

# Output library technical names
string(TOLOWER ${LIBRARY_NAME} LIBRARY_NAME_LC)
set(LIBRARY_BIN_NAME "mdt${LIBRARY_NAME_LC}")
# Add output library
add_library(${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX} SHARED ${SOURCES_FILES})
target_link_libraries(${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX} "mdterrorOld${MDTLIB_VERSION_SUFFIX}")
target_link_libraries(${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX} "mdterrorguiOld${MDTLIB_VERSION_SUFFIX}")
target_link_libraries(${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX} "mdtqtsingleapplication${MDTLIB_VERSION_SUFFIX}")

# Translation files
set(TS_FILES
    ${CMAKE_CURRENT_SOURCE_DIR}/translations/${LIBRARY_BIN_NAME}_fr.ts
    ${CMAKE_CURRENT_SOURCE_DIR}/translations/${LIBRARY_BIN_NAME}_de.ts
)
qt5_create_translation(QM_FILES ${CMAKE_CURRENT_SOURCE_DIR}/src ${TS_FILES} OPTIONS -I ${CMAKE_CURRENT_SOURCE_DIR}/src)
add_custom_target("Mdt${LIBRARY_NAME}Ts" DEPENDS ${QM_FILES} ${TS_FILES})

# Headers to install
file(GLOB INSTALL_HEADERS *.h)
install(FILES ${INSTALL_HEADERS} DESTINATION "${HEADERS_DEST_DIR}/Mdt/${LIBRARY_NAME}" COMPONENT dev)

# Install targets
install(TARGETS "${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX}" DESTINATION "${LIB_DEST_DIR}" COMPONENT lib)

# Strip library and put debug symbols into separate file
if(UNIX)
  add_custom_command(TARGET "${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX}"
                    POST_BUILD
                    COMMAND objcopy --only-keep-debug "${CMAKE_SHARED_LIBRARY_PREFIX}${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX}.so" "${CMAKE_SHARED_LIBRARY_PREFIX}${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX}.dbg"
                    COMMAND strip --strip-debug --strip-unneeded "${CMAKE_SHARED_LIBRARY_PREFIX}${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX}.so"
                    COMMAND objcopy --add-gnu-debuglink="${CMAKE_SHARED_LIBRARY_PREFIX}${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX}.dbg" "${CMAKE_SHARED_LIBRARY_PREFIX}${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX}.so"
                    COMMAND chmod -x "${CMAKE_SHARED_LIBRARY_PREFIX}${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX}.dbg")
  install(FILES "${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX}.dbg" DESTINATION "${LIB_DEST_DIR}" COMPONENT dbg)
endif(UNIX)

# Unit tests
add_subdirectory("tests")
