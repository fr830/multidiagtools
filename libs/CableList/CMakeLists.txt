# Project file for Multi Diag Tools
# * Part: CableList *

# Project
project(multidiagtools)

# List of source files
set(SOURCES_FILES 
    src/Mdt/CableList/DatabaseSchema.cpp
    src/Mdt/CableList/ModificationModel.cpp
    src/Mdt/CableList/LinkTypeModel.cpp
    src/Mdt/CableList/Path/Graph.cpp
)

# Headers directories
include_directories("${CMAKE_CURRENT_SOURCE_DIR}/src")
# Because some headers are generated during building (config.h, Qt's UIC), we also build path
include_directories("${CMAKE_CURRENT_BINARY_DIR}")

# Output library
add_library(mdtcablelist${MDTLIB_VERSION_SUFFIX} SHARED ${SOURCES_FILES})
target_link_libraries(mdtcablelist${MDTLIB_VERSION_SUFFIX} "mdterror${MDTLIB_VERSION_SUFFIX}")
target_link_libraries(mdtcablelist${MDTLIB_VERSION_SUFFIX} "mdterrorgui${MDTLIB_VERSION_SUFFIX}")
target_link_libraries(mdtcablelist${MDTLIB_VERSION_SUFFIX} "mdtdatabase${MDTLIB_VERSION_SUFFIX}")

# Translation files
set(TS_FILES
    ${CMAKE_CURRENT_SOURCE_DIR}/translations/mdtcablelist_fr.ts
    ${CMAKE_CURRENT_SOURCE_DIR}/translations/mdtcablelist_de.ts
)
qt5_create_translation(QM_FILES ${CMAKE_CURRENT_SOURCE_DIR}/src ${TS_FILES} OPTIONS -I ${CMAKE_CURRENT_SOURCE_DIR}/src)
add_custom_target(MdtCableListTs DEPENDS ${QM_FILES} ${TS_FILES})

# Headers to install
file(GLOB INSTALL_HEADERS *.h)
install(FILES ${INSTALL_HEADERS} DESTINATION "${HEADERS_DEST_DIR}/mdtcablelist" COMPONENT dev)

# Install targets
install(TARGETS "mdtcablelist${MDTLIB_VERSION_SUFFIX}" DESTINATION "${LIB_DEST_DIR}" COMPONENT lib)

# Strip library and put debug symbols into separate file
if(UNIX)
  add_custom_command(TARGET "mdtcablelist${MDTLIB_VERSION_SUFFIX}"
                    POST_BUILD
                    COMMAND objcopy --only-keep-debug "libmdtcablelist${MDTLIB_VERSION_SUFFIX}.so" "libmdtcablelist${MDTLIB_VERSION_SUFFIX}.dbg"
                    COMMAND strip --strip-debug --strip-unneeded "libmdtcablelist${MDTLIB_VERSION_SUFFIX}.so"
                    COMMAND objcopy --add-gnu-debuglink="libmdtcablelist${MDTLIB_VERSION_SUFFIX}.dbg" "libmdtcablelist${MDTLIB_VERSION_SUFFIX}.so"
                    COMMAND chmod -x "libmdtcablelist${MDTLIB_VERSION_SUFFIX}.dbg")
  install(FILES "libmdtcablelist${MDTLIB_VERSION_SUFFIX}.dbg" DESTINATION "${LIB_DEST_DIR}" COMPONENT dbg)
endif(UNIX)

# Unit tests
add_subdirectory("tests")
