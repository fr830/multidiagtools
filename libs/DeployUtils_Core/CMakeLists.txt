
# TODO
# On Windows, we must have all required libraries available
# in the same directory than executables (tests, final tools)
# The final goal is that our tools makes the majority of the work.
#
# Steps:
#  - Copy needed functions from MdtDependenciesUtils.cmake here
#  - Copy needed libs (can do it by hand, 
#     with expericnce we will know what to copy for which compiler/arch)
#  - Once all runs good (hope it will happen some day..), update functions in MdtDependenciesUtils.cmake
#    and also the main CMakeLists.txt .

add_subdirectory(src)
add_subdirectory(tests)
add_subdirectory(translations)
