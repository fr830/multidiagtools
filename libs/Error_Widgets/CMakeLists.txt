# Project file for Mdt Error_Widgets

add_subdirectory(src)
add_subdirectory(tests)
add_subdirectory(translations)
