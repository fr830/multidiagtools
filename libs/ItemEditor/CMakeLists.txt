# Project file for Multi Diag Tools

# Project
project(multidiagtools)

# List of source files
set(SOURCES_FILES
    src/Mdt/ItemEditor/Debug.cpp
    src/Mdt/ItemEditor/RowState.cpp
    src/Mdt/ItemEditor/ItemDelegateProxy.cpp
    src/Mdt/ItemEditor/EventCatchItemDelegate.cpp
    src/Mdt/ItemEditor/ItemSelectionModel.cpp
    src/Mdt/ItemEditor/ControllerState.cpp
    src/Mdt/ItemEditor/ControllerEvent.cpp
    src/Mdt/ItemEditor/RowChangeEventDispatcher.cpp
    src/Mdt/ItemEditor/AbstractActionContainer.cpp
    src/Mdt/ItemEditor/NavigationActions.cpp
    src/Mdt/ItemEditor/InsertAction.cpp
    src/Mdt/ItemEditor/RemoveAction.cpp
    src/Mdt/ItemEditor/EditionActions.cpp
    src/Mdt/ItemEditor/ResizeToContentsAction.cpp
    src/Mdt/ItemEditor/AbstractControllerStatePermission.cpp
    src/Mdt/ItemEditor/ControllerStatePermissionProxyModel.cpp
    src/Mdt/ItemEditor/ControllerStateTableRow.cpp
    src/Mdt/ItemEditor/AbstractControllerStateTable.cpp
    src/Mdt/ItemEditor/ControllerStateMachine.cpp
    src/Mdt/ItemEditor/AbstractController.cpp
    src/Mdt/ItemEditor/ItemViewPrivateContainer.cpp
    src/Mdt/ItemEditor/TableViewControllerImplementation.cpp
    src/Mdt/ItemEditor/TableViewControllerStatePermission.cpp
    src/Mdt/ItemEditor/TableViewControllerStateTable.cpp
    src/Mdt/ItemEditor/TableViewController.cpp
    src/Mdt/ItemEditor/MappedWidgetList.cpp
    src/Mdt/ItemEditor/WidgetEditablePropertyMap.cpp
    src/Mdt/ItemEditor/WidgetStyleSheet.cpp
    src/Mdt/ItemEditor/DataWidgetMapperItemDelegate.cpp
    src/Mdt/ItemEditor/DataWidgetMapper.cpp
    src/Mdt/ItemEditor/WidgetMapperControllerStatePermission.cpp
    src/Mdt/ItemEditor/WidgetMapperControllerStateTable.cpp
    src/Mdt/ItemEditor/WidgetMapperController.cpp
    src/Mdt/ItemEditor/AbstractControllerRelation.cpp
    src/Mdt/ItemEditor/FilterControllerRelation.cpp
    src/Mdt/ItemEditor/NavigationControllerRelation.cpp
    src/Mdt/ItemEditor/ControllerRelationList.cpp
    src/Mdt/ItemEditor/StandardEditorLayoutWidget.cpp
    
    src/Mdt/ItemEditor/AbstractEditorLayout.cpp
    src/Mdt/ItemEditor/AbstractEditorWidget.cpp
    
    src/Mdt/ItemEditor/AbstractFormWidget.cpp
    src/Mdt/ItemEditor/AbstractTableViewWidget.cpp
    src/Mdt/ItemEditor/TableViewWidget.cpp
    src/Mdt/ItemEditor/StandardWindow.cpp
    src/Mdt/ItemEditor/AbstractWindow.cpp
)

# Headers directories
include_directories("${CMAKE_CURRENT_SOURCE_DIR}/src")
include_directories("${CMAKE_CURRENT_SOURCE_DIR}/../FilterExpression/src")
include_directories("${CMAKE_CURRENT_SOURCE_DIR}/../ItemModel/src")
# Because some headers are generated during building (config.h, Qt's UIC), we also build path
include_directories("${CMAKE_CURRENT_BINARY_DIR}")

# Output library name
set(LIBRARY_NAME "ItemEditor")

# Output library technical names
string(TOLOWER ${LIBRARY_NAME} LIBRARY_NAME_LC)
set(LIBRARY_BIN_NAME "mdt${LIBRARY_NAME_LC}")
# Add output library
add_library(${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX} SHARED ${SOURCES_FILES})
target_link_libraries(${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX} "mdterror${MDTLIB_VERSION_SUFFIX}")
target_link_libraries(${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX} "mdterrorgui${MDTLIB_VERSION_SUFFIX}")
target_link_libraries(${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX} "mdtfilterexpression${MDTLIB_VERSION_SUFFIX}")
target_link_libraries(${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX} "mdtitemmodel${MDTLIB_VERSION_SUFFIX}")
target_link_libraries(${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX} Qt5::Widgets)

# Qt Designer plugins
add_subdirectory("qttools")

# Translation files
set(TS_FILES
    ${CMAKE_CURRENT_SOURCE_DIR}/translations/${LIBRARY_BIN_NAME}_fr.ts
    ${CMAKE_CURRENT_SOURCE_DIR}/translations/${LIBRARY_BIN_NAME}_de.ts
)
qt5_create_translation(QM_FILES ${CMAKE_CURRENT_SOURCE_DIR}/src ${TS_FILES} OPTIONS -I ${CMAKE_CURRENT_SOURCE_DIR}/src)
add_custom_target("Mdt${LIBRARY_NAME}Ts" DEPENDS ${QM_FILES} ${TS_FILES})

# Headers to install
file(GLOB INSTALL_HEADERS *.h)
install(FILES ${INSTALL_HEADERS} DESTINATION "${HEADERS_DEST_DIR}/Mdt/${LIBRARY_NAME}" COMPONENT dev)

# Install targets
install(TARGETS "${LIBRARY_BIN_NAME}${MDTLIB_VERSION_SUFFIX}" DESTINATION "${LIB_DEST_DIR}" COMPONENT lib)

# Strip library and put debug symbols into separate file
if(UNIX)
  mdtSeparateDebugSymbols(${LIBRARY_BIN_NAME})
endif(UNIX)

# Unit tests
add_subdirectory("tests")
# Examples
add_subdirectory("examples")
