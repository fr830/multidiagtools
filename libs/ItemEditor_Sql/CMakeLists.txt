
add_subdirectory(src)
add_subdirectory(tests)
add_subdirectory(examples)
add_subdirectory(translations)
