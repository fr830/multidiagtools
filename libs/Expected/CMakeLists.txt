# Project file for Multi Diag Tools

# Project
project(multidiagtools)

# Headers directories
include_directories("${CMAKE_CURRENT_SOURCE_DIR}/src")

# Output library name
set(LIBRARY_NAME "Expected")

# Output library technical names
string(TOLOWER ${LIBRARY_NAME} LIBRARY_NAME_LC)
set(LIBRARY_BIN_NAME "mdt${LIBRARY_NAME_LC}")

# Headers to install
file(GLOB INSTALL_HEADERS *.h)
install(FILES ${INSTALL_HEADERS} DESTINATION "${HEADERS_DEST_DIR}/Mdt/${LIBRARY_NAME}" COMPONENT dev)

# Unit tests
add_subdirectory("tests")
